web/src/saved_replies.ts

```diff
@@ -0,0 +1,161 @@
+import type {StateData} from "./state_data";
+import * as util from "./util";
+
+export type SavedReplies = {
+    id: number;
+    title: string;
+    content: string;
+    rendered_content: string;
+    date_created: number;
+};
+
+const ADD_SAVED_REPLY_OPTION_ID = -1;
+let my_saved_replies: SavedReplies[] = [];
+
+export function set_saved_replies(saved_replies: SavedReplies[]): void {
+    my_saved_replies = saved_replies;
+    my_saved_replies.sort((a, b) => util.strcmp(a.title.toLowerCase(), b.title.toLowerCase()));
+}
+
+export function get_saved_replies(): SavedReplies[] {
+    return my_saved_replies;
+}
+
+export function add_saved_reply(saved_reply: SavedReplies): void {
+    my_saved_replies.push(saved_reply);
+    my_saved_replies.sort((a, b) => util.strcmp(a.title.toLowerCase(), b.title.toLowerCase()));
+}
+
+export function remove_saved_reply(saved_reply_id: number): void {
+    my_saved_replies = my_saved_replies.filter((s) => s.id !== saved_reply_id);
+}
+
+export function get_options_for_dropdown_widget(): Option[] {
+    const saved_replies = my_saved_replies.map((saved_reply) => ({
+        unique_id: saved_reply.id,
+        name: saved_reply.title,
+        description: saved_reply.rendered_content.replace(/^(?:<p>)?(.*?)(?:<\/p>)?$/, "$1"),
+        rendered_description: true,
+        bold_current_selection: true,
+        is_saved_reply: true,
+    }));
+
+    // Option for creating a new saved reply.
+    saved_replies.unshift({
+        unique_id: ADD_SAVED_REPLY_OPTION_ID,
+        name: $t({defaultMessage: "Add a new saved reply"}),
+        description: "",
+        rendered_description: false,
+        bold_current_selection: true,
+        is_saved_reply: false,
+    });
+    return saved_replies;
+}
+
+function submit_create_saved_reply_form(): void {
+    const title = $<HTMLInputElement>("#add-new-saved-reply-modal .title").val()?.trim();
+    const content = $<HTMLInputElement>("#add-new-saved-reply-modal .content").val()?.trim();
+    if (title && content) {
+        dialog_widget.submit_api_request(channel.post, "/json/saved_replies", {title, content});
+    }
+}
+
+function update_submit_button_state(): void {
+    const title = $<HTMLInputElement>("#add-new-saved-reply-modal .title").val()?.trim();
+    const content = $<HTMLInputElement>("#add-new-saved-reply-modal .content").val()?.trim();
+    const $submit_button = $("#add-new-saved-reply-modal .dialog_submit_button");
+
+    $submit_button.prop("disabled", true);
+    if (title && content) {
+        $submit_button.prop("disabled", false);
+    }
+}
+
+function saved_reply_modal_post_render(): void {
+    $("#add-new-saved-reply-modal").on("input", "input,textarea", update_submit_button_state);
+
+    $("#add-new-saved-reply-modal").on("keypress", "input", (event) => {
+        if (keydown_util.is_enter_event(event)) {
+            event.preventDefault();
+            submit_create_saved_reply_form();
+        }
+    });
+}
```

Comment:
Hmm, does `dialog_widget` not provide a way to configure enter-submits without special code like this?

Reply:
Yeah, I noticed it does when `form_id` is provided in the constructor. Updated it now.